webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/action/impl/mail/MailParserAction.java

```diff
@@ -0,0 +1,125 @@
+package won.bot.framework.eventbot.action.impl.mail;
+
+import com.hp.hpl.jena.rdf.model.Model;
+import org.apache.commons.lang3.StringUtils;
+import won.bot.framework.eventbot.EventListenerContext;
+import won.bot.framework.eventbot.action.EventBotActionUtils;
+import won.bot.framework.eventbot.action.impl.needlifecycle.AbstractCreateNeedAction;
+import won.bot.framework.eventbot.event.Event;
+import won.bot.framework.eventbot.event.impl.mail.MailReceivedEvent;
+import won.bot.framework.eventbot.event.impl.wonmessage.FailureResponseEvent;
+import won.bot.framework.eventbot.listener.EventListener;
+import won.protocol.message.WonMessage;
+import won.protocol.model.BasicNeedType;
+import won.protocol.model.FacetType;
+import won.protocol.service.WonNodeInformationService;
+import won.protocol.util.NeedModelBuilder;
+import won.protocol.util.RdfUtils;
+import won.protocol.util.WonRdfUtils;
+
+import javax.mail.Address;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * Created by fsuda on 30.09.2016.
+ */
+public class MailParserAction extends AbstractCreateNeedAction {
+    private String uriMailRelationsName;
+
+    public MailParserAction(EventListenerContext eventListenerContext, String uriListName, String uriMailRelationsName, URI... facets) {
+        super(eventListenerContext, uriListName);
+        this.uriMailRelationsName = uriMailRelationsName;
+
+        this.usedForTesting = true;
+
+        if (facets == null || facets.length == 0) {
+            //add the default facet if none is present.
+            this.facets = new ArrayList<URI>(1);
+            this.facets.add(FacetType.OwnerFacet.getURI());
+        } else {
+            this.facets = Arrays.asList(facets);
+        }
+    }
+
+    protected void doRun(Event event) throws Exception {
+        logger.info("CREATENEEDFROMMAILACTION IS CALLED");
```

Comment:
we should use 'info' loglevel more sparingly than that. Here, it might be ok, but further down (subject, ...) it's too fine-grained. And we should refrain from all-caps logging. It's rude.